src/Sructs.c

```diff
@@ -0,0 +1,40 @@
+/*
+ Created by Elchairoy Meir on 17/11/2021.
+ Definition of table, column a cell structs.
+*/
+
+#define MAX_COLUMN_NAME_LEN 30
+#define MAX_TABLE_NAME_LEN 30
+
+enum types {Character = 0,Integer,Double,String}; /* All the possible types of data in each column/cell. */
+
+/* The struct of cells: */
+typedef struct
+{
+    enum types type;
+    unsigned index; /* The index of the cell in the initial column (after the column goes through changes that
+                        might not be the number of the cell in the column). */
+    union {
+        char char_value;
+        int int_value;
+        double double_value;
+        char *string_value;
+    }value /* The value of the cell (can be several types). */;
+}cell;
+
+/* The struct of columns: */
+typedef struct
+{
+    enum types type;
+    char name[MAX_COLUMN_NAME_LEN]; /* The name of the column. */
+    unsigned number_of_cells; /* The number of cells in the column. */
```

Comment:
Like in the cell struct - should be much bigger than unisgned! You can have billions of cells!